framework_tvb/tvb/adapters/simulator/simulator_fragments.py

```diff
@@ -273,16 +267,23 @@ def fill_from_post(all_range_parameters, **data):
         if pse_param2_name:
             pse_param2 = all_range_parameters.get(pse_param2_name)
             if pse_param2.type is float:
-                pse_param2_lo = data.get('_pse_param2_lo')
-                pse_param2_hi = data.get('_pse_param2_hi')
-                pse_param2_step = data.get('_pse_param2_step')
+                pse_param2_lo = data.get('pse_param2_lo')
+                pse_param2_hi = data.get('pse_param2_hi')
+                pse_param2_step = data.get('pse_param2_step')
                 param2_range = RangeParameter(pse_param2_name, float,
                                               Range(float(pse_param2_lo), float(pse_param2_hi), float(pse_param2_step)),
                                               is_array=pse_param2.is_array)
             else:
-                param2_range_str = data.get('_pse_param2')
+                param2_range_str = data.get('pse_param2')
                 pse_param2_name += '.gid'
                 param2_range = RangeParameter(pse_param2_name, pse_param1.type,
                                               SimulatorPSEParamRangeFragment._prepare_pse_uuid_list(param2_range_str))
 
         return param1_range, param2_range
+
+
+class SimulatorPSEFinalFragment(ABCAdapterForm):
```

Comment:
Can you pls explain why for PSE we need a different final form for the name ?
Can't we use the one with the length as on regular single simulations ?

Reply:
That field should be simulation_number, not name... I dunno why I wrote simulation_name. I added that final form because there was a TO-DO which required to separate the setting of range params from the pse_launching and the only thing the last fragment should do is dispay the number of simulations that are about to be launched.

Reply:
Actually Paula suggested me to use a form for that and I think the idea was that the user could see the number of simulations to be launched before actually launching them (like a last chance to change his mind if he thinks that there are too many or not enough). But if that's not important, I think I could display it from JS and in that case the launch_pse method will just get a None parameter for the fragment.

Reply:
As discussed today, this SimulatorPSEFinalFragment could disappear, and we shall use the JS code for displaying and computing the message with PSE no of ops.